### PR TITLE
Do not pip install openstack client, use regular SUSE packages

### PR DESCRIPTION
**Pull request by jsuchome**
Provide custom 010-setup-client.sh that does do installation,
only the configration part.

### CODE REVIEW

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 1]
```diff
@@ -130,8 +130,25 @@
     - horizon
     - run
 
+- name: Install openstack client
+  package:
+    name: python-openstackclient
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: could you convert that to a list with two packages?
- python-openstackclient
- python-heatclient

As the latter has a dependency on the former, not the other way around, IIRC.

File: playbooks/roles/deploy-osh/files/010-setup-client.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      username: 'admin'
+      password: 'password'
+      project_name: 'admin'
+      project_domain_name: 'default'
+      user_domain_name: 'default'
+      auth_url: 'http://keystone.openstack.svc.cluster.local/v3'
+EOF
+
+#NOTE: Build charts
+make all
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/files/010-setup-client.sh]: I would prefer if we make all the charts explicitly, rather than in here. We are building the osh-infra charts in main anyway, it would be just nice to have the osh charts built afterwards, and more explicit. The airship squad could add their repos too.

Comment (evrardjp) [reply playbooks/roles/deploy-osh/files/010-setup-client.sh]: That's more a nit than a problem for me, if we keep the file. Just saying :)

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 1 above)
Comment (dirkmueller) [reply playbooks/roles/deploy-osh/tasks/main.yml]: heatclient certainly does not depend on openstackclient, its the other way, aka the way jiri did it.

File: playbooks/roles/deploy-osh/files/010-setup-client.sh (hunk 2 above)
Comment (jsuchome) [reply playbooks/roles/deploy-osh/files/010-setup-client.sh]: OK

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 3]
```diff
@@ -130,17 +130,44 @@
...
+    owner: root
+    group: root
+    mode: 0755
+  tags:
+    - run
+
+- name: Save default clouds.yaml file
+  copy:
+    srt: clouds.yaml
+    dest: /etc/openstack/clouds.yaml
```
Comment (dirkmueller) [inline playbooks/roles/deploy-osh/tasks/main.yml]: why do we need to install that system wide rather than in the home directory?

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: I guess we don't really _need_ it. This is just copying the functionality of helm's `010-setup-client.sh`

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: That said, I think I could put it under `~/.config/openstack/` , unless there really is some reason to have it in global location.

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 1 above)
Comment (evrardjp) [reply playbooks/roles/deploy-osh/tasks/main.yml]: that's interesting, I didn't know openstackclient bundled everything in this packaging. I double checked and it's the case indeed.

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 3 above)
Comment (evrardjp) [reply playbooks/roles/deploy-osh/tasks/main.yml]: I think both are fine. Let's use home, it seems cleaner :)
We'll see if it eventually conflicts with the expectations of other things upstream too, and we could improve upstream then.

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 4]
```diff
@@ -130,17 +130,44 @@
...
+    path: /etc/openstack
+    state: directory
+    owner: root
+    group: root
+    mode: 0755
+  tags:
+    - run
+
+- name: Save default clouds.yaml file
+  copy:
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: I would use template: here, so that we can change it later with a few overrides, should that path be pursued.

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 5]
```diff
@@ -130,17 +130,44 @@
...
+    srt: clouds.yaml
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: I think that's `src:` here

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: yep, i've spotted it as well

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 4 above)
Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: ok, I'll adapt it

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 1 above)
Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: It is not bundled in one package, it's brought by dependency

File: playbooks/roles/deploy-osh/tasks/main.yml (hunk 3 above)
Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: OK

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 6]
```diff
@@ -130,17 +130,43 @@
     - horizon
     - run
 
+- name: Install openstack client
+  package:
+    name: python-openstackclient
+    state: present
+  register: _pkginstall
```
Comment (Itxaka) [inline playbooks/roles/deploy-osh/tasks/main.yml]: do we really need all this stuff with retries? This should only be used if we expect issues but why would we expect issues on a simple package install?

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: usually not, but it could theoretically happen that some zypper process is already running...

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: So I'm just using same values I've found elsewhere

Comment (Itxaka) [reply playbooks/roles/deploy-osh/tasks/main.yml]: shouldnt then we use the zypper module with `environment:
    ZYPP_LOCK_TIMEOUT: 20` as options to suse-ify it more??

Comment (Itxaka) [reply playbooks/roles/deploy-osh/tasks/main.yml]: substitute 20 with the wanted value of course :)

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/main.yml]: Might be good idea to do extra PR that would replace all package installations in our plays

Comment (evrardjp) [reply playbooks/roles/deploy-osh/tasks/main.yml]: > do we really need all this stuff with retries? This should only be used if we expect issues but why would we expect issues on a simple package install?

network related issues. That's why we brought that into ansible-lint.
ansible-lint will complain if you don't bring that.

Let's not make it extra hard with locks, just make it retries, so we don't have to change that everywhere. If we need to change the module to have this ZYPP_LOCK_TIMEOUT by default, we can too.

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 7]
```diff
@@ -130,17 +130,43 @@
...
-    executable: /bin/bash
+- name: Create .config/openstack directory
+  file:
+    path: /root/.config/openstack
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+  tags:
+    - run
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: nit: technically that can still be install.

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 8]
```diff
@@ -130,17 +130,43 @@
...
+
+- name: Save default clouds.yaml file
+  template:
+    src: clouds.yaml.j2
+    dest: /root/.config/openstack/clouds.yaml
+    owner: root
+    mode: '0640'
+
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: nit: while this should be tagged `run`.

File: playbooks/roles/deploy-osh/tasks/main.yml [hunk 9]
```diff
@@ -130,17 +130,43 @@
...
+- name: Build openstack-helm charts
+  make:
     chdir: /opt/openstack-helm
+    target: all
+  when:
+    - not developer_mode
+    - _gitclone is changed
```
Comment (evrardjp) [inline playbooks/roles/deploy-osh/tasks/main.yml]: that should probably go away. If developer mode is used, the charts need to be built anyway, and the clone would not be changed. But that can be in a different commit.
